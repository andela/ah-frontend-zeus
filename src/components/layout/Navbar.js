import React, { Component } from 'react';
import { Link } from 'react-router-dom';
/**
 * randers the navigation bar.
 */
class Navbar extends Component {
  render() {
    return (
      <nav className="navbar navbar-expand-sm navbar-dark bg-dark mb-5 fixed-top">
        <div className="container">
          <Link to="/" className="navbar-brand">
            Authors Haven
          </Link>
          <button
            type="button"
            data-toggle="collapse"
            data-target="#mobile-nav"
            className="navbar-toggler"
          >
            <span className="navbar-toggler-icon" />
          </button>
          <div className="collapse navbar-collapse" id="mobile-nav">
            <ul className="navbar-nav ml-auto">
              <li className="nav-item">
                <Link to="/login" className="nav-link mr-2">
                  Sign in
                </Link>
              </li>
              <li className="nav-item">
                <Link
                  to="/register"
                  className="nav-link btn btn-outline-secondary"
                >
                  Sign up
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    );
  }
}

export default Navbar;
