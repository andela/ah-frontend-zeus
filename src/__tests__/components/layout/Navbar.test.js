import React from 'react';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import {shallow } from 'enzyme';
import Navbar from '../../../components/layout/Navbar';
import { Link } from 'react-router-dom';


Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('<Navbar/>', () => {
  it('should render three <Navbar/> elements', () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper.find(Link)).toHaveLength(3);
  
  });
  
});

it('testing Nav bar using snapshot', () => {
  const wrapper = shallow(<Navbar />);
  expect(wrapper).toMatchSnapshot();

});




