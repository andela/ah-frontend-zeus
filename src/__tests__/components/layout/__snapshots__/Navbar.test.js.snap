// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Nav bar using snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <Link
          className="navbar-brand"
          replace={false}
          to="/"
        >
          Authors Haven
        </Link>
        <button
          className="navbar-toggler"
          data-target="#mobile-nav"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="mobile-nav"
        >
          <ul
            className="navbar-nav ml-auto"
          >
            <li
              className="nav-item"
            >
              <Link
                className="nav-link mr-2"
                replace={false}
                to="/login"
              >
                Sign in
              </Link>
            </li>
            <li
              className="nav-item"
            >
              <Link
                className="nav-link btn btn-outline-secondary"
                replace={false}
                to="/register"
              >
                Sign up
              </Link>
            </li>
          </ul>
        </div>
      </div>,
      "className": "navbar navbar-expand-sm navbar-dark bg-dark mb-5 fixed-top",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="navbar-brand"
            replace={false}
            to="/"
          >
            Authors Haven
          </Link>,
          <button
            className="navbar-toggler"
            data-target="#mobile-nav"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>,
          <div
            className="collapse navbar-collapse"
            id="mobile-nav"
          >
            <ul
              className="navbar-nav ml-auto"
            >
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link mr-2"
                  replace={false}
                  to="/login"
                >
                  Sign in
                </Link>
              </li>
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link btn btn-outline-secondary"
                  replace={false}
                  to="/register"
                >
                  Sign up
                </Link>
              </li>
            </ul>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Authors Haven",
            "className": "navbar-brand",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Authors Haven",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="navbar-toggler-icon"
            />,
            "className": "navbar-toggler",
            "data-target": "#mobile-nav",
            "data-toggle": "collapse",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "navbar-toggler-icon",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="navbar-nav ml-auto"
            >
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link mr-2"
                  replace={false}
                  to="/login"
                >
                  Sign in
                </Link>
              </li>
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link btn btn-outline-secondary"
                  replace={false}
                  to="/register"
                >
                  Sign up
                </Link>
              </li>
            </ul>,
            "className": "collapse navbar-collapse",
            "id": "mobile-nav",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link mr-2"
                    replace={false}
                    to="/login"
                  >
                    Sign in
                  </Link>
                </li>,
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link btn btn-outline-secondary"
                    replace={false}
                    to="/register"
                  >
                    Sign up
                  </Link>
                </li>,
              ],
              "className": "navbar-nav ml-auto",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link mr-2"
                    replace={false}
                    to="/login"
                  >
                    Sign in
                  </Link>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sign in",
                    "className": "nav-link mr-2",
                    "replace": false,
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": "Sign in",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link btn btn-outline-secondary"
                    replace={false}
                    to="/register"
                  >
                    Sign up
                  </Link>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sign up",
                    "className": "nav-link btn btn-outline-secondary",
                    "replace": false,
                    "to": "/register",
                  },
                  "ref": null,
                  "rendered": "Sign up",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <Link
            className="navbar-brand"
            replace={false}
            to="/"
          >
            Authors Haven
          </Link>
          <button
            className="navbar-toggler"
            data-target="#mobile-nav"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>
          <div
            className="collapse navbar-collapse"
            id="mobile-nav"
          >
            <ul
              className="navbar-nav ml-auto"
            >
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link mr-2"
                  replace={false}
                  to="/login"
                >
                  Sign in
                </Link>
              </li>
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link btn btn-outline-secondary"
                  replace={false}
                  to="/register"
                >
                  Sign up
                </Link>
              </li>
            </ul>
          </div>
        </div>,
        "className": "navbar navbar-expand-sm navbar-dark bg-dark mb-5 fixed-top",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="navbar-brand"
              replace={false}
              to="/"
            >
              Authors Haven
            </Link>,
            <button
              className="navbar-toggler"
              data-target="#mobile-nav"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>,
            <div
              className="collapse navbar-collapse"
              id="mobile-nav"
            >
              <ul
                className="navbar-nav ml-auto"
              >
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link mr-2"
                    replace={false}
                    to="/login"
                  >
                    Sign in
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link btn btn-outline-secondary"
                    replace={false}
                    to="/register"
                  >
                    Sign up
                  </Link>
                </li>
              </ul>
            </div>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Authors Haven",
              "className": "navbar-brand",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Authors Haven",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="navbar-toggler-icon"
              />,
              "className": "navbar-toggler",
              "data-target": "#mobile-nav",
              "data-toggle": "collapse",
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "navbar-toggler-icon",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul
                className="navbar-nav ml-auto"
              >
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link mr-2"
                    replace={false}
                    to="/login"
                  >
                    Sign in
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link btn btn-outline-secondary"
                    replace={false}
                    to="/register"
                  >
                    Sign up
                  </Link>
                </li>
              </ul>,
              "className": "collapse navbar-collapse",
              "id": "mobile-nav",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="nav-link mr-2"
                      replace={false}
                      to="/login"
                    >
                      Sign in
                    </Link>
                  </li>,
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="nav-link btn btn-outline-secondary"
                      replace={false}
                      to="/register"
                    >
                      Sign up
                    </Link>
                  </li>,
                ],
                "className": "navbar-nav ml-auto",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="nav-link mr-2"
                      replace={false}
                      to="/login"
                    >
                      Sign in
                    </Link>,
                    "className": "nav-item",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Sign in",
                      "className": "nav-link mr-2",
                      "replace": false,
                      "to": "/login",
                    },
                    "ref": null,
                    "rendered": "Sign in",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="nav-link btn btn-outline-secondary"
                      replace={false}
                      to="/register"
                    >
                      Sign up
                    </Link>,
                    "className": "nav-item",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Sign up",
                      "className": "nav-link btn btn-outline-secondary",
                      "replace": false,
                      "to": "/register",
                    },
                    "ref": null,
                    "rendered": "Sign up",
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
